aten/src/ATen/native/sparse/cuda/SparseCsrTensorMath.cu

```diff
@@ -117,6 +118,61 @@ void convert_indices_from_csr_to_coo_cuda(const Tensor& indices, const Tensor& c
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
+// This function unfolds the compressed indices of a compressed sparse matrix
+// into a batched compressed sparse tensor.
+// This is analogous to an unflatten-like operation:
+// unflatten(0, {b, r}) for csr/bsr with input shape (r*b, c)
+//          (output shape (b, r, c))
+// unflatten(1, {b, c}).transpose(0,1) for csc/bsc with input shape (r, c*b)
+//          (output shape (r, b, c) unflatten, (b, r, c) unflatten + transpose)
+// This only operates on the compressed indices as the plain indices and values
+// can be manipulated as described above without special handling.
+// It is a prerequisite for the conversion that the sparsity pattern is sane for
+// the batched shape. That is each batch has the same number of nonzero
+// elements.
+template <typename input_t, typename output_t>
+__global__ void compressed_to_batched_compressed_indices_cuda_kernel(
+    output_t* __restrict__ batched_out,
+    const input_t* __restrict__ compressed_in,
+    const int64_t n_batch,
+    const int64_t ncomp_per_batch) {
+  int64_t b = blockDim.x * blockIdx.x + threadIdx.x;
+  const int64_t batch_length = ncomp_per_batch + 1;
+  if (b < n_batch) {
+    const auto c0 = b * ncomp_per_batch;
+    //Subtract the first cindx value from each within the batch to get the batched_cindx value.
+    const auto batch_delta = compressed_in[c0];
+    const auto b0 = b * batch_length;
+    for (int64_t i = 0; i < batch_length; ++i) {
+      batched_out[b0 + i] =
+          static_cast<output_t>(compressed_in[c0 + i] - batch_delta);
+    }
+  }
+}
+
+template <typename input_t, typename output_t>
+void compressed_to_batched_compressed_indices_cuda(
+    const Tensor& batched_out,
+    const Tensor& compressed_in,
+    int64_t n_batch) {
+  auto compressed_ = compressed_in.expect_contiguous();
+  TORCH_INTERNAL_ASSERT(batched_out.is_contiguous());
+  input_t* compressed_data_in = compressed_->data_ptr<input_t>();
+  output_t* batched_data_out = batched_out.data_ptr<output_t>();
+  const auto ncomp_per_batch = batched_out.size(-1) - 1;
+
+  // Run n_batch threads...
+  int64_t THREADS = at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock;
+  int64_t BLOCKS = (n_batch + THREADS) / THREADS;
```

Comment:
So, if we keep the kernel, shouldn't it better be `(n_batch + THREADS - 1) / THREADS`?